pkg/controllers/awsloadbalancercontroller/controller.go

```diff
@@ -218,20 +246,43 @@ func (r *AWSLoadBalancerControllerReconciler) SetupWithManager(mgr ctrl.Manager)
 		Owns(&appsv1.Deployment{}).
 		Owns(&corev1.Service{}).
 		Owns(&arv1.ValidatingWebhookConfiguration{}).
-		Owns(&arv1.MutatingWebhookConfiguration{}).
-		Complete(r)
+		Owns(&arv1.MutatingWebhookConfiguration{})
+
+	if r.TrustedCAConfigMapName != "" {
+		clusterALBCInstance := func(o client.Object) []reconcile.Request {
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Name: controllerName,
+					},
+				},
+			}
+		}
+
+		bldr = bldr.Watches(&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(clusterALBCInstance),
```

Comment:
I need some help on this, will come back to it.

Reply:
How about adding some comments here?

Reply:
Shouldn't you check for changes first before you requeue it?

Reply:
What changes do you have in mind?

The logic here is to re-trigger the awsloadbalancercontroller's reconciliation loop when the controller runtime sees any changes on the trusted CA configmap we specified in the predicates. The reconciliation loop aligns the operands with the current state. The trusted CA configmap is a part of the alignment - we calculate its checksum and see if it needs a new rollout of the managed deployment.

Reply:
Update: I read [this comment](https://github.com/openshift/aws-load-balancer-operator/pull/90#discussion_r1144992436) and I guess I know where the confusion comes from. The trusted CA configmap on its own is not part of the desired state, but the consumption of this configmap is. The operator has to make sure a volume and volumeMount is added to the managed aws lb controller's deployment. The contents of the configmap though are opaque to the operator, it makes the managed deployment consume whatever is in there.

Reply:
I'm going to ask someone else to review this.